File: result/translate/starchat/java2cpp/c++/WAYS_TO_WRITE_N_AS_SUM_OF_TWO_OR_MORE_POSITIVE_INTEGERS.cpp
int f_gold ( int n ) {
    vector <int> table (n + 1, 0);
    for (int i = 0; i <= n; i++) table[i] = 1;
    for (int i = 1; i < n; i++) {
        for (int j = i; j <= n; j++) table[j] += table[j - i];
    }
    return table[n];
}

File: dataset/python2java/python/WAYS_TO_WRITE_N_AS_SUM_OF_TWO_OR_MORE_POSITIVE_INTEGERS.py
def f_gold ( n ) :
    table = [ 0 ] * ( n + 1 )
    table [ 0 ] = 1
    for i in range ( 1 , n ) :
        for j in range ( i , n + 1 ) :
            table [ j ] += table [ j - i ]
    return table [ n ]